packages/flutter_tools/gradle/flutter.gradle

```diff
@@ -409,9 +414,24 @@ class FlutterPlugin implements Plugin<Project> {
 
         // Wait until the Android plugin loaded.
         pluginProject.afterEvaluate {
+            Boolean showBuildConfigurationMessage = false
+
+            // Checks if there is a mismatch between the plugin compileSdkVersion and the project compileSdkVersion.
             if (pluginProject.android.compileSdkVersion > project.android.compileSdkVersion) {
                 project.logger.quiet("Warning: The plugin ${pluginName} requires Android SDK version ${pluginProject.android.compileSdkVersion.substring(8)}.")
+                showBuildConfigurationMessage = true
             }
+
+            // Checks if there is a mismatch between the plugin minSdkVersion and the project minSdkVersion.
+            if(pluginProject.android.defaultConfig?.minSdkVersion?.apiLevel > project.android.defaultConfig?.minSdkVersion?.apiLevel) {
```

Comment:
this is handled by the tool down below too, right?
In this case, I'd suggest to remove this code from flutter.gradle the reduce bugs in the future.